clang/docs/ReleaseNotes.rst

```diff
@@ -148,6 +148,19 @@ C++ Specific Potentially Breaking Changes
     // Now diagnoses with an error.
     void f(int& i [[clang::lifetimebound]]);
 
+- Clang will now prefer the ``[[nodiscard]]`` declaration on function declarations over ``[[nodiscard]]``
```

Comment:
I'm not sure this is really a 'potentially breaking change' so much as just a diagnostic change?